new recipe for booster v0.1.2

BenjaminDoran opened the pull request.

giordano commented on B/booster/build_tarballs.jl:
```diff
@@ -0,0 +1,41 @@
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd ${WORKSPACE}/srcdir/booster
+install_license ${WORKSPACE}/srcdir/booster/LICENSE
+
```
The license should be installed automatically, if you want you can remove all of this
```suggestion
```

giordano commented on B/booster/build_tarballs.jl:
```diff
@@ -0,0 +1,41 @@
+
+# compile booster
+cd ${WORKSPACE}/srcdir/booster/src
+make -j${nproc}
```
I don't really like their default choices.  We want to use clang on BSD systems:
```suggestion
make -j${nproc} CC=${CC} CFLAGS_OMP="-Wall -g -fopenmp"
```

giordano commented on B/booster/build_tarballs.jl:
```diff
@@ -0,0 +1,41 @@
+
+# Dependencies that must be installed before this package can be built
+dependencies = Dependency[
+    Dependency("OpenMPI_jll", v"4.1.2"),
```
Is MPI used at all?  Doesn't look like it?

BenjaminDoran replied:
The binary needs the potential to be used with MPI, i.e. the binary has an argument for number of cpu's to use, and needs to be compiled with the MPI flags available.

if there is a best practice to get this effect I am open to it.

giordano replied:
I don't see any reference at all to MPI in the code. Are you sure you aren't confused with OpenMP, which, as confusing as it is, it's something completely different?

giordano commented on B/booster/build_tarballs.jl:
```diff
@@ -0,0 +1,41 @@
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = Dependency[
```
```suggestion
dependencies = [
    Dependency(PackageSpec(name="CompilerSupportLibraries_jll", uuid="e66e0078-7015-5450-92f7-15fbd957f2ae"); platforms=filter(!Sys.isbsd, platforms)),
    Dependency(PackageSpec(name="LLVMOpenMP_jll", uuid="1d63c593-3942-5779-bab2-d838dc0a180e"); platforms=filter(Sys.isbsd, platforms)),
```

BenjaminDoran commented:
Yeah (super confusing), I think you are right. I added a new commit that
removed this dependency. Testing locally it compiled and worked correctly.
It is the latest commit in the pull request which also passed.

-Benjamin


On Mon, Mar 28, 2022 at 3:34 PM Mosè Giordano ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In B/booster/build_tarballs.jl
> <https://github.com/JuliaPackaging/Yggdrasil/pull/4685#discussion_r836826572>
> :
>
> > +# move to bindir
> +install -Dvm 755 "booster" "${bindir}/booster${exeext}"
> +"""
> +
> +# These are the platforms we will build for by default, unless further
> +# platforms are passed in on the command line
> +platforms = supported_platforms()
> +
> +# The products that we will ensure are always built
> +products = [
> +    ExecutableProduct("booster", :booster),
> +]
> +
> +# Dependencies that must be installed before this package can be built
> +dependencies = Dependency[
> +    Dependency("OpenMPI_jll", v"4.1.2"),
>
> I don't see any reference at all to MPI in the code. Are you sure you
> aren't confused with OpenMP, which, as confusing as it is, it's something
> completely different?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/JuliaPackaging/Yggdrasil/pull/4685#discussion_r836826572>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AE4LRXQYUTIJQ77WGSBBJ3DVCIJWVANCNFSM5R37OMZQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
